### PR TITLE
Test framework to run JLP tests

### PR DESCRIPTION
**Pull request by schemar**
Added a test framework that connects to ethereum nodes, runs the tests
against them, and exits.
Run tests with `npm run test`.

Use the `shared` module in a test to access the connection and the chain config.
A "global" `afterEach` hook automatically updates the configuration in between
runs so that all tests can access the same shared configuration file.
The test configuration file is expected to be the same format as the
standard JLP configuration file.

Anchor deployment is just an example for easier adoption. It shows how
to access shared objects.

The tests use a separate configuration file. It is recreated every time
you run `npm run test`. It is copied from `./test/config_init.json`
where you need to set the basic parameters like web3 endpoints.

It currently usese the standard accounts from JLP. Once we have a
faucet, the tests should create and remove accounts specific to each
test run.

### CODE REVIEW

**Comment by schemar** [comment]
I forgot something! The current implementation gives the error that we load babel/polyfill multiple times and we shouldn't. I am guessing it has something to do with loading the dependencies, but I am not sure. Any expert on this who can help?

File: README.md [hunk 1]
```diff
@@ -233,6 +233,14 @@ It will write `tokenRulesAddress` to your config file.
 * Replace `eip20Token` with eip20Token address.
 * Replace `organization` with an organization contract address. 
 
+## Tests
+
+To run the tests:
+
+1. Copy `./test/config_init.json.dist` to `./test/config_init.json`
```
Comment (pgev) [inline README.md]: I would propose to automate it as much as possible.
We could insist that for running tests the values in config are there and valid.

File: test/connected.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const { assert } = require('chai');
+
+const ChainConfig = require('../src/config/chain_config');
+const Connection = require('../src/connection');
+
+const { CONFIG_FILE_PATH } = require('./const');
```
Comment (pgev) [inline test/connected.js]: Confusing :)

File: test/connected.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ *         });
+ *       },
+ *     );
+ *   });
+ * });
+ *
+ * @param {function} execute The callback that will be executed once the connection is established.
+ */
+const run = async (execute) => {
+  let connection;
```
Comment (pgev) [inline test/connected.js]: let connection = null;

File: test/anchor/deploy.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const connected = require('../connected');
+const Deployer = require('../../src/deployer');
+
+describe('Anchor', async () => {
+  it('deploys anchors', async () => {
+    await connected.run(
```
Comment (pgev) [inline test/anchor/deploy.js]: Let's have connection as shared module or global to easy writing tests.

**Pull request merged**